src/posts/interning-at-khan-academy.md

```diff
@@ -0,0 +1,53 @@
+
+# Part 1: ramping up
+
+My first day at Khan Academy was a whirlwind of activities. From setting up my laptop to having a 1:1 with my mentor and finally adding myself to the [interns page](https://www.khanacademy.org/about/our-interns), I had a lot of things to do. As soon as I settled down with my first deploy, I got to work on the XOM initiative, starting with improving the math keypad.
+
+As part of the XOM initiative, the team had designed a math keypad that could be used to type in expression responses to mathematics problems in a mobile-friendly way. I started by fixing some bugs to get familiar with the math keypad codebase, such as disabling gestures on the static sidebar, a fixed bar on the side of the keypad with navigation buttons. Once I had a good idea of how the keypad worked, I was able to start working on adding new features our designers had crafted. I worked on things such as using square icons for representing empty blocks, and learned in the process how important it is to make these small changes to create a great experience for students.
+
+At Khan Academy, we use React.js, something I had worked with before, but only in the context of [Scala.js](https://www.scala-js.org) applications. As part of making changes to the math keypad, I had to learn how to write idiomatic ES6 code. Working on Khan Academy gave me a great opportunity to use React with ES6 in a production environment, and left me with many ideas on how React.js could be used more seamlessly with Scala.js. In addition to learning React and ES6, I got a chance to work with the plethora of open-source libraries at Khan Academy such as [Aphrodite](https://github.com/khan/aphrodite) and [KaTeX](https://github.com/khan/katex).
+
+![changes to the math keypad](/images/interning-at-khan-academy/image_0.png)
+
+*an early screenshot of changes to the math keypad*
+
+# Part 2: iframes, iframes, iframes
+
+Now that I was used to the flow of picking a task, creating a code review, and deploying my changes, I started on my first project -- improving our content editor to be mobile first. This involved pulling together specs for what types of devices students use Khan Academy on, rendering content with the new styles the rest of the team was working on, isolating the rendered content to prevent styles leaking from the rest of the editor, and implementing touch emulation for interacting with the mobile previews.
```

Comment:
Style nits:

* Let's use a colon instead of a dash here, too :)

Stray thoughts:

* What does "mobile first" mean in this context? My first interpretation was that we rewrote the content editor to be usable on mobile devices, but I think you probably mean that we want our tooling to encourage content producers to build content mobile-first, right?
  * Maybe we should start by explaining that this was the problem we were trying to solve: "…my first project: updating our CMS to support mobile-first content creation. Historically, our content creators have worked primarily on desktop: they write their content on desktop, they preview their content on desktop, they review each other's work on desktop, they check their email on desktop. How can we make it easy for our content creators to build content for mobile first, and thereby avoid awful small-screen layout bugs [maybe a cute screenshot link here], but still allow them to take advantage of the big-screen multi-tasking perks of desktop devices?"
* On that note, a nit: in terms of, like, verb tenses, it's probably slightly better storytelling for this paragraph to describe the problems you had to overcome, instead of the solutions you enacted: "How do we ensure that editor styles don't leak into the content preview? How do we simulate touchscreens on non-touch devices?" That sorta thing :) Low-priority; it's just cute.
* Why was it helpful to know the common device types? Was that just to ensure that our manual QA had good device coverage, or did it actually affect implementation somehow? I'm not _sure_ this needs explained in the article, exactly; mostly I'm just curious :)